.concourse/scripts/t-cargo-build.sh

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env bash
+
+set -e
+set -x
+set -o pipefail
+
+GIT_DIR=$1
+CACHE_DIR=$2
+ARTIFACT_DIR=$3
+SEMVER_DIR=$4
+
+SEMVER=$(cat $SEMVER_DIR/version)
+GIT_REF=""
+if [[ -f ${GIT_DIR}/.git/short_ref ]]; then
+    GIT_REF=$(cat ${GIT_DIR}/.git/short_ref)
+elif [[ -f "${GIT_DIR}/.git/resource/head_sha" ]]; then
+    GIT_REF=$(cat ${GIT_DIR}/.git/resource/head_sha)
+else
+    echo "no reference for the polymesh version found"
+    exit 1
+fi
+
+# Prime the build cache directory
+mkdir -p "${CACHE_DIR}"
+if [ ! -f ".git/resource/changed_files" ] || grep -v '^.concourse\|^Dockerfile\|^scripts/cli' ".git/resource/changed_files" || [ ! -f "target/release/polymesh" ]; then
+    touch ${CACHE_DIR}/.new_binary
+else
+    rm -f ${CACHE_DIR}/.new_binary
+fi
+
+pushd .
+cd $GIT_DIR
+# Compile if any of the following conditions is met:
+#  - This is a branch merge, not a PR
+#  - This is a PR where the source minus exceptions changed
+#  - The polymesh binary is missing
+if [ ! -f ".git/resource/changed_files" ] || grep -v '^.concourse\|^Dockerfile\|^scripts/cli' ".git/resource/changed_files" || [ ! -f "target/release/polymesh" ]; then
+    rm -f target/release/polymesh
+    cargo +$TOOLCHAIN build --release || cargo build -j 1 --release
```

Comment:
`+$TOOLCHAIN` is not required. Cargo will automatically use the toolchain mentioned in the `.toolchain` file or fallback to default toolchain, which is the latest installed stable toolchain.

Reply:
Does it hurt to have it here?  Where is the `.toolchain` file located?

Reply:
`+$TOOLCHAIN` overrides `.toolchain`. I think it's better to respect `.toolchain` everywhere for consistency.

Reply:
What happens when `.toolchain` specifies a version of Rust that is not available?  In our containers `$TOOLCHAIN` is guaranteed to be the same version as the nightly that we add. Alternatively I could rename `$TOOLCHAIN` to `$RUSTUP_TOOLCHAIN`.

Reply:
It is automatically downloaded. The thing I'm not sure about is if the downloading is handled by Cargo (installed on our CI) or rustup (not installed on our CI). Perhaps something we can test.